### DISCUSSION TITLE
[UX] allow insertion of connected subpatch in connection via ctrl+t

### DISCUSSION THREAD

**Discussion by ben-wes** [pure-data/pure-data#2370] [Ideas]
maybe this could directly be added as issue - but i'm noting it here for now:

Pd allows to create a connected subpatch from a selected object - automagically adding an inlet to the newly created subpatch in the upper left corner.

this is not the case when adding an object in a connection via <kbd>ctrl</kbd>+<kbd>t</kbd> though (where the connections are lost). in that case, it would obviously need to create an inlet and outlet to preserve connections.

... maybe this is a feature that would be more appropriate as part of a vanilla patcherize integration though?

**Comment by porres** [comment]
I don't understand, can you give pics and more details? I am lost

**Comment by ben-wes** [comment]
sure ... here's what currently happens:

1. ![Screenshot 2024-07-01 at 17 48 37](https://github.com/pure-data/pure-data/assets/1923787/4d76d10d-1682-4439-9b4a-61270cd98110)
2. ![Screenshot 2024-07-01 at 17 49 12](https://github.com/pure-data/pure-data/assets/1923787/466cdc34-1a60-4891-8060-56f69144c493)
3. ![Screenshot 2024-07-01 at 17 49 17](https://github.com/pure-data/pure-data/assets/1923787/9b87aa76-43eb-49e6-9c06-572810daa2c8)
4. ![Screenshot 2024-07-01 at 17 48 57](https://github.com/pure-data/pure-data/assets/1923787/63f4cc96-149b-4e53-9454-bd15080ef024)

here's what the result could be for 3 and 4 according to the suggested changes:

3. ![Screenshot 2024-07-01 at 17 50 53](https://github.com/pure-data/pure-data/assets/1923787/12546d76-eec9-472c-a183-aa9cfcaf3a45)
4. ![Screenshot 2024-07-01 at 17 49 57](https://github.com/pure-data/pure-data/assets/1923787/10740fd2-dd23-4e4c-8f2e-5de4af88ad51)

**Reply by umlaeute** [reply]
i quite like the idea of re-using existing connections of an object that is replaced by a subpatch. (as opposed to the `[pd nop~]` that is inserted when triggerizing a signal cord).

the question how to deal with something like this
![gnurz~](https://github.com/pure-data/pure-data/assets/214034/146fe220-276a-44a3-9435-de1aef55a0e0)

should the result mimick all iolets of the original object:

```
[inlet~]    [inlet~]    [inlet~]


[outlet~]   [outlet~]   [inlet~]
```

or just the connected ones?

```
[inlet~]    [inlet~]


[outlet~]   [outlet~]
```

**Reply by ben-wes** [reply]
in this case, i don't really have a strong preference (at least on a first thought). it seems acceptable to me to only consider the existing connections. the additional ones can still be added manually at any position without loosing connections.

and yes, actually the case i describe there for a single connection is an equivalent for `[pd nop~]` in the control domain - the in- and outlet could even be preconnected in the subpatch. not sure about the ideal behaviour, i admit.

**Reply by umlaeute** [reply]
sidenote: i'm somewhat inclined to change the `[pd nop~]` in signal cords to just `[+~]`.
together with *this* suggestion (as in #2372)  we than have
- consistent mimimal behaviour for inserting an object into a cord
- a generalized approach to substituting a single object with a subpatch

**Comment by porres** [comment]
I see, thanks
